src/demo/providers/LogoutProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, useEffect, useState } from "react";
+import { api, ExternalUser, SdkError, SdkErrors } from "@tonomy/tonomy-id-sdk";
+import useErrorStore from "../../common/stores/errorStore";
+
+interface LogoutContextProps {
+  logout: () => void;
+}
+
+interface LogoutProviderProps {
+  children: React.ReactNode;
+}
+
+export const LogoutContext = createContext<LogoutContextProps>({
+  logout: () => {},
+});
+
+const LogoutProvider: React.FC<LogoutProviderProps> = ({ children }) => {
```

Comment:
maybe this needs to be a more generic "UserProvider" which manages the User state relevant for routes

With something like this...
```ts
interface UserContextProps {
  login(user: ExternalUser) => void
  logout: () => void;
  user?: ExternalUser
}
```

Reply:
like this
https://github.com/remix-run/react-router/blob/dev/examples/auth/src/App.tsx#L74